src/index.js

```diff
@@ -5,52 +5,68 @@ const SCRIPT_ID = 'hcaptcha-api-script-id';
 const HCAPTCHA_LOAD_FN_NAME = 'hcaptchaOnLoad';
 
 // Prevent loading API script multiple times
-let resolveFn;
-let rejectFn;
-const mountPromise = new Promise((resolve, reject) => {
-  resolveFn = resolve;
-  rejectFn = reject;
-});
+const scripts = [];
 
 // Generate hCaptcha API script
 const mountCaptchaScript = (params={}) => {
   const parent = params.scriptLocation || document.head;
+
   delete params.scriptLocation;
 
   const doc = parent.ownerDocument || document;
+  const win = doc.defaultView || doc.parentWindow;
+
+  const script = scripts.find(({ scope }) => scope === win);
 
-  if (doc.getElementById(SCRIPT_ID)) {
+  if (doc.getElementById(SCRIPT_ID) && script) {
     // API was already requested
-    return mountPromise;
+    return script.promise;
   }
 
-  // Create global onload callback
-  window[HCAPTCHA_LOAD_FN_NAME] = resolveFn;
+  const promise = new Promise((resolve, reject) => {
+    // Create global onload callback
+    win[HCAPTCHA_LOAD_FN_NAME] = resolve;
+
+    const domain = params.apihost || "https://js.hcaptcha.com";
+    delete params.apihost;
 
-  const domain = params.apihost || "https://js.hcaptcha.com";
-  delete params.apihost;
+    const script = doc.createElement("script");
+    script.id = SCRIPT_ID;
+    script.src = `${domain}/1/api.js?render=explicit&onload=${HCAPTCHA_LOAD_FN_NAME}`;
 
-  const script = doc.createElement("script");
-  script.id = SCRIPT_ID;
-  script.src = `${domain}/1/api.js?render=explicit&onload=${HCAPTCHA_LOAD_FN_NAME}`;
+    script.async = params.loadAsync !== undefined? params.loadAsync : true;
+    delete params.loadAsync;
 
-  script.async = params.loadAsync !== undefined? params.loadAsync : true;
-  delete params.loadAsync;
+    script.onerror = (event) => reject('script-error');
 
-  script.onerror = (event) => rejectFn('script-error');
+    const query = generateQuery(params);
+    script.src += query !== ""? `&${query}` : "";
```

Comment:
missing whitespace `query !== "" ? ...`

Reply:
@faris-imi I can follow up with this update in a separate PR